lib/npg_pipeline/function/autoqc.pm

```diff
@@ -207,47 +188,34 @@ sub _create_definition4interop {
 
   $ref->{'command'}  = $c;
 
-  return npg_pipeline::function::definition->new($ref);
+  return $self->create_definition($ref);
 }
 
 sub _create_definition_object {
   my ($self, $product, $command) = @_;
 
-  my $ref = $self->_basic_attrs();
+  my $ref = {};
   my $qc_to_run = $self->qc_to_run;
 
   $ref->{'job_name'}        = $self->_job_name();
-  $ref->{'fs_slots_num'}    = 1;
   $ref->{'composition'}     = $product->composition;
   $ref->{'command'}         = $command;
 
   if ($qc_to_run eq q[adapter]) {
-    $ref->{'num_cpus'}      = [$self->general_values_conf()->{'qc_adapter_cpu'} || 1];
-    if ($ref->{'num_cpus'} > 1) {
-      $ref->{'num_hosts'}   = 1;
-    }
-  }
-
-  $ref->{'apply_array_cpu_limit'} = 1;
-  #####
-  # Lower value for ref_match to try to alleviate Lustre client multiple
-  # simulaneous access bug (ensure elements only run eight at a time).
-  #
-  if ($qc_to_run eq 'ref_match') {
-    $ref->{'array_cpu_limit'} = $REFMATCH_ARRAY_CPU_LIMIT;
+    $ref->{num_cpus} = [$self->general_values_conf()->{'qc_adapter_cpu'}] || [1];
```

Comment:
The number of CPUs for this job will now come from the function graph, this assignment is not needed.